pkg/kopia/repositoryserver/config/cr.kanister.io_repositoryservers.yaml

```diff
@@ -0,0 +1,233 @@
+
+---
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    controller-gen.kubebuilder.io/version: v0.4.1
+  creationTimestamp: null
+  name: repositoryservers.cr.kanister.io
+spec:
+  group: cr.kanister.io
+  names:
+    kind: RepositoryServer
+    listKind: RepositoryServerList
+    plural: repositoryservers
+    singular: repositoryserver
+  scope: Namespaced
+  versions:
+  - name: v1alpha1
+    schema:
+      openAPIV3Schema:
+        description: RepositoryServer manages the lifecycle of Kopia Repository Server
+          within a Pod
+        properties:
+          apiVersion:
+            description: 'APIVersion defines the versioned schema of this representation
+              of an object. Servers should convert recognized schemas to the latest
+              internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources'
+            type: string
+          kind:
+            description: 'Kind is a string value representing the REST resource this
+              object represents. Servers may infer this from the endpoint the client
+              submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds'
+            type: string
+          metadata:
+            type: object
+          spec:
+            description: RepositoryServerSpec is the specification for the RepositoryServer
+            properties:
+              location:
+                description: Location references the backend store where a repository
+                  already exists and the credential necessary to connect to the backend
+                  store
+                properties:
+                  credentialSecretName:
```

Comment:
We discussed adding a `SecretReference` object type with `name` and `namespace` for these secrets. Could you please change the definition to reflect that?

Reply:
@ihcsim @pavannd1 Can we rename the `Location` object to `Storage` object in this CR? This `Location` struct is conflicting with the already present Location field in Profile struct from types.go

Calling it `Storage` is in congruence with what @ankitjain235 is doing in his repository wrapper PR https://github.com/kanisterio/kanister/pull/1648